api/basic/add-pin.ts

```diff
@@ -0,0 +1,178 @@
+import { DynamoDBClient, ConditionalCheckFailedException } from '@aws-sdk/client-dynamodb'
+import { DynamoDBDocumentClient, GetCommand, PutCommand } from '@aws-sdk/lib-dynamodb'
+import { SQSClient, SendMessageCommand } from '@aws-sdk/client-sqs'
+import { APIGatewayProxyEventV2 } from 'aws-lambda'
+import { ClusterAddResponse, Pin, Response } from './schema.js'
+import { CID } from 'multiformats/cid'
+import { Multiaddr } from 'multiaddr'
+import { nanoid } from 'nanoid'
+
+interface UpsertPinInput {
+  cid: string
+  dynamo: DynamoDBClient
+  table: string
+}
+
+interface AddToQueueInput {
+  cid: string
+  origins: string[]
+  bucket: string
+  sqs: SQSClient
+  queueUrl: string
+}
+
+// type AddPinInput = AddToQueueInput & UpsertPinInput
+interface AddPinInput extends UpsertPinInput, AddToQueueInput {}
+
+/**
+ * AWS API Gateway handler for POST /pin/${cid}?&origins=${multiaddr},${multiaddr}
+ * Collect the params and delegate to addPin to do the work
+ *
+ * We provide responses in Payload format v2.0
+ * see: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.proxy-format
+ */
+export async function handler (event: APIGatewayProxyEventV2): Promise<Response> {
+  const {
+    TABLE_NAME: table = '',
+    BUCKET_NAME: bucket = '',
+    QUEUE_URL: queueUrl = '',
+    CLUSTER_BASIC_AUTH_TOKEN: token = '',
+    // set for testing
+    SQS_ENDPOINT: sqsEndpoint = undefined,
+    DYNAMO_DB_ENDPOINT: dbEndpoint = undefined
+  } = process.env
+
+  if (event.headers.authorization !== `Basic ${token}`) {
+    return { statusCode: 401, body: { error: { reason: 'UNAUTHORIZED' } } }
+  }
+
+  const sqs = new SQSClient({ endpoint: sqsEndpoint })
+  const dynamo = new DynamoDBClient({ endpoint: dbEndpoint })
+  const cid = event.pathParameters?.cid ?? ''
```

Comment:
I wouldn't default this to '' and adopt null instead. Then it will fail in `addPin`

Reply:
I'm trying out ts-standard on this project, and it won't let me do coercing if checks like if(someVar) unless someVar is actually a boolean, hence all this unidiomatic oddness. 

I can flip it to null, but both null and '' are checked for in addPin and the cid property is typed as string, so I'm using `''` as the as the only possible falsy value here. I agree it all feels a bit awkward tho.